src/app/Dashboard/AutomatedAnalysis/AutomatedAnalysisConfigForm.tsx

```diff
@@ -65,60 +61,43 @@ import {
   TextVariants,
   ValidatedOptions,
 } from '@patternfly/react-core';
-import { CogIcon } from '@patternfly/react-icons';
 import * as React from 'react';
-import { Link } from 'react-router-dom';
+import { useTranslation } from 'react-i18next';
 import { concatMap, filter, first } from 'rxjs';
 interface AutomatedAnalysisConfigFormProps {
-  onCreate?: () => void;
-  onSave?: () => void;
   isSettingsForm: boolean;
 }
 
-export const AutomatedAnalysisConfigForm: React.FC<AutomatedAnalysisConfigFormProps> = ({
-  onCreate,
-  onSave,
-  ...props
-}) => {
+export const AutomatedAnalysisConfigForm: React.FC<AutomatedAnalysisConfigFormProps> = ({ ...props }) => {
   const context = React.useContext(ServiceContext);
   const addSubscription = useSubscriptions();
+  const [t] = useTranslation();
 
+  const parseEventString = (): [string | undefined, TemplateType | undefined] => {
+    const eventString = context.settings.automatedAnalysisRecordingConfig().template;
+    if (!eventString) {
+      return [undefined, undefined];
+    }
+    const templateName = eventString.split(',')[0].split('=')[1];
+    const templateType = eventString.split(',')[1].split('=')[1];
+    if (!(templateType === 'TARGET' || templateType === 'CUSTOM')) {
+      console.error(`Invalid template type ${templateType}`);
+      return [undefined, undefined];
+    }
+    return [templateName, templateType];
+  };
+  const [recordingConfig, setRecordingConfig] = React.useState<AutomatedAnalysisRecordingConfig>(
+    context.settings.automatedAnalysisRecordingConfig()
+  );
   const [templates, setTemplates] = React.useState([] as EventTemplate[]);
-  const [templateName, setTemplateName] = React.useState<string | undefined>(undefined);
-  const [templateType, setTemplateType] = React.useState<TemplateType | undefined>(undefined);
-  const [maxAge, setMaxAge] = React.useState(context.settings.automatedAnalysisRecordingConfig().maxAge);
+  const [templateName, setTemplateName] = React.useState<string | undefined>(parseEventString()[0]);
```

Comment:
Since we already have the `recordingConfig` which holds all the config. We can remove these states (i.e. `templateName`, `maxAge`, `maxSize`? We will still need to keep the unit tho.